Properly restoring tester status methods

milljm opened the pull request:
This is work required to make a proper --failed-tests DAG operation

A more efficient skippedDependency checker

Modify the QueueManager to utilize the same status restoring as
--failed-tests techniques.

Closes #13424

Commit:
Properly restoring tester status methods

This is work required to make a proper --failed-tests DAG operation

A more efficient skippedDependency checker

Modify the QueueManager to utilize the same status restoring as
--failed-tests techniques.

Closes #13424

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/340212/) on 7fe93ce wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/13437/site)

This comment will be updated on new commits.

permcody commented:
I just tested this locally and it doesn't work at all. `--failed-tests` appears completely broken. No matter what I did it always reran all of the tests when I used that flag.

Testing procedure: Modify "simple_transient_diffusion:BCs/right/value=2"
```
./run_tests --re=simple
./run_tests --failed-tests
```

# To test the prerequisite part of this PR
```
./run_tests --re=simple --recover
./run_tests --failed-tests
```

That last test case is the one I'm interested in. I would expect a failure in the Part2 for simple_transient_diffusion.i, when rerunning I'd like it to rerun part1 and part2. Right now, it only attempts to run part2 but since there is a prerequisite on part2, it just skips it instead and runs zero tests.

Commit:
Fixes --failed-tests running all tests

WIP: Still trying to figure out how to make --recover work. The issue,
is that when --recover is used, the Factory artificially creates testers
that don't exist and adds a prereq that otherwise also did not exist to
the original tester. When it comes time to use --failed-tests, this
stateful information is lost, and thus the new run has no idea what the
previous dependency tree looked like.

permcody reviewed (approved):
Looks good - not sure if I should merge or let Sam get his stuff.
